notebooks/article_1.py

```diff
@@ -0,0 +1,1093 @@
+        },
+        {
+            "x": "age_last_funding_year",
+            "y": "age_last_milestone_year",
+            "hue": "negative_last_milestone",
+            "xlabel": "Age at Last Funding Year",
+            "ylabel": "Age at Last Milestone Year",
+            "data": data_df
+        },
+        {
+            "x": "diff_founded_first_funding",
+            "y": "age_first_funding_year",
+            "hue": "negative_first_funding",
+            "xlabel": "Difference 'Founded' and 'First Funding'",
+            "ylabel": "Age at First Funding Year",
+            "data": data_df  # Specify the DataFrame
+        },
+        {
+            "x": "diff_founded_last_funding",
+            "y": "age_last_funding_year",
+            "hue": "negative_last_funding",
+            "xlabel": "Difference 'Founded' and 'Last Funding'",
+            "ylabel": "Age at Last Funding Year",
+            "data": data_df
+        }
+
+
+    ]
+
+    # Generate each scatter plot based on the configuration
+    for i, config in enumerate(plots_config, start=1):
+        plt.subplot(1, 4, i)
+        sns.scatterplot(
+            x=config["x"],
+            y=config["y"],
+            hue=config["hue"],
+            palette=custom_palette,
+            data=config["data"]
+        )
+        plt.xlabel(config["xlabel"])
+        plt.ylabel(config["ylabel"])
+
+    plt.tight_layout()  # Adjust the layout
+    plt.show()
+
+outliers_analysis(dataframe)
+
+
+# From the **scatterplot** we identified the following:
+#
+# - Negative values are highlighted in red, while positive values are indicated in blue.
+#
+# - The occurrence of negative values could be attributed to several reasons, such a data entry errors, or a relative timing. However, without definitive context, we cannot be certain of the exact cause.
+#
+# For the purposes of our analysis, we'll focus on how to handle these negative values. By doing so, we can maintain the integrity of our analysis and ensure that our insights are based on accurate and representative data.
+
+# #### Handle Negative Values
+
+# We will now use `np.abs` to take the absolute value of the input.
+
+# In[605]:
+
+
+dataframe["age_first_funding_year"]=np.abs(dataframe["age_first_funding_year"])
+dataframe["age_last_funding_year"]=np.abs(dataframe["age_last_funding_year"])
+dataframe["age_first_milestone_year"]=np.abs(dataframe["age_first_milestone_year"])
+dataframe["age_last_milestone_year"]=np.abs(dataframe["age_last_milestone_year"])
+
+
+# Let's observe how the plot changed after removing negative values.
+
+# In[606]:
+
+
+def identify_outliers(data_df):
+    threshold = 4
+    column_name = "age_first_funding_year"
+    column = data_df[column_name]
+    mean = column.mean()
+    std = column.std()
+    is_outlier = (np.abs(column - mean) > threshold * std)
+    data_df['is_outlier'] = is_outlier.astype(int)
+    return data_df
+
+dataframe = identify_outliers(dataframe)
+
+def plot_outlier_scatterplots(data_df):
```

Comment:
Is there similarity with `outliers_scatter_plot` declared above instead?
Would it be worth to simplify?